### PR TITLE
Drop unittest2 in favor of newer unittest implementations

### PR DESCRIPTION
**Pull request by FedericoRessi**
unittest2 library isn't been updated since some year.

Newer python frameworks are expected to has far more fixes than that unittest2.

Actually unittest2 documentation states it is only tested up to python 3.4

This therefore limit the use of unittest2 library up to the last python 3.4 release.

It is expected to fix Issue #235 when recent enough version of Python is installed

### CODE REVIEW

**Comment by stmcginnis** [comment]
I'm all for this change, but I think it will take a little more work.

There are places in the code that expect unittest2 to be present:

https://github.com/testing-cabal/testtools/blob/83cb1545e0b409e3cf7fbab6eff7a333713822cd/testtools/testsuite.py#L38

With this change, unittest2 will only be installed in old python versions. Newer versions will have an issue at runtime when it tries to load classes like these.

We also limit installation to python 3.5 or great:

https://github.com/testing-cabal/testtools/blob/83cb1545e0b409e3cf7fbab6eff7a333713822cd/setup.py#L14

So this makes it kind of a no-op.

I think ideally we should phase out the use of unitttest2. There are other issues like https://github.com/testing-cabal/testtools/issues/272#issuecomment-573860489 where the consuming code needs to handle the appropriate unittest or unittest2 code. So if we change this, we need to make sure it is released with a major version bump so it isn't entirely unexpected in consuming projects if things start breaking for them.

**Comment by mtreinish** [comment]
Yeah, I agree with @stmcginnis here, we've moved to only supporting python 3.5, I think the best path forward here is to just use unittest everywhere and remove all the unittest2 usage. This is in progress (albeit slowly because I keep getting sidetracked with other work) in #277

**Comment by stmcginnis** [comment]
We should probably stick with https://github.com/testing-cabal/testtools/pull/277

**Comment by FedericoRessi** [comment]
@mtreinish I merged #277 into this to get what I was missing. Please merge this into your #277 so you get what you missed.
If you look well in run.py you still have a reference to unittest2 around line 47. Cheers

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,18 @@
+[tox]
+envlist = py35,py36,pypy,pypy3
```
Comment (hugovk) [inline tox.ini]: Python 2 is no longer supported, and 3.7 and 3.8 are supported.

```suggestion
envlist = py35,py36,py37,py38,pypy3
```

**Comment by mtreinish** [comment]
I've pulled this into #277 so I'm going to close this

**Pull request closed without merging**